firewood/src/proof.rs

```diff
@@ -89,6 +89,33 @@ pub struct ProofNode {
     pub child_hashes: Children<HashType>,
 }
 
+impl std::fmt::Debug for ProofNode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let key: Vec<u8> = self.key().collect();
+        let children: Vec<(usize, TrieHash)> = self
+            .children()
+            .iter()
+            .enumerate()
+            .filter_map(|(i, c)| c.as_ref().map(|h| (i, (*h).clone().into_triehash())))
```

Comment:
I think this should probably be `IntoHashType::into_hash_type()` instead of `into_triehash`. We've sort of make TrieHash specific to storage and HashType the generic for the node id. 

But, it already is a `HashType` so probably don't need it at all. 

`Clone` in a debug impl is also a red flag. everything should work by reference.